tensorflow/lite/micro/compression/view.py

```diff
@@ -0,0 +1,424 @@
+  for index, t in enumerate(tensors):
+    d = {
+        "_tensor": index,
+        "name": _decode_name(t.name),
+        "type": unpack_TensorType(t.type),
+        "shape": t.shape,
+        "buffer": t.buffer,
+    }
+
+    if t.isVariable:
+      d["is_variable"] = True
+    else:
+      # don't display this unusual field
+      pass
+
+    if t.quantization is not None and t.quantization.scale is not None:
+      d["quantization"] = {
+          "scale": t.quantization.scale,
+          "zero": t.quantization.zeroPoint,
+          "dimension": t.quantization.quantizedDimension,
+      }
+    result.append(d)
+
+    if codec is not None:
+      coordinates = TensorCoordinates(subgraph_ix=subgraph_index,
+                                      tensor_index=index)
+      d |= unpack_compression(coordinates, codec)
+
+  return result
+
+
+def unpack_compression(tensor: TensorCoordinates, codec: Codec) -> dict:
+  result = {}
+
+  compressions = codec.list_compressions(tensor)
+  if compressions:
+    result["_compressed"] = [c.name for c in compressions]
+    metadata = codec._tensor_metadata(tensor)
+    assert metadata is not None
+    result["_value_buffer"] = metadata.valueBuffer
+    result["_lookup_tables"] = codec.lookup_tables(tensor)
+
+  return result
+
+
+def unpack_subgraphs(model: tflite_schema.ModelT, codec: Codec | None):
+  result = []
+  for index, s in enumerate(model.subgraphs):
+    d = {
+        "_subgraph": index,
+        "_operator_count": len(s.operators),
+        "_tensor_count": len(s.tensors),
+        "name": _decode_name(s.name),
+        "operators": unpack_operators(model, s.operators),
+        "tensors": unpack_tensors(s.tensors, subgraph_index=index,
+                                  codec=codec),
+    }
+    result.append(d)
+  return result
+
+
+def unpack_opcodes(opcodes: list[tflite_schema.OperatorCodeT]) -> list:
+  result = []
+  for index, opcode in enumerate(opcodes):
+    d: dict = {
+        "_opcode_index": index,
+        "_name": OPERATOR_NAMES[opcode.builtinCode],
+        "builtin_code": opcode.builtinCode,
+        "version": opcode.version,
+    }
+    if opcode.customCode is not None:
+      d["custom_code"] = opcode.customCode
+      del d["_name"]
+    result.append(d)
+  return result
+
+
+def unpack_metadata(model: tflite_schema.ModelT):
+  entries = []
+  compression = MetadataReader.build(model)
+
+  if model.metadata is None:
+    return entries
+
+  for m in model.metadata:
+    d = {"name": _decode_name(m.name), "buffer": m.buffer}
+
+    if compression and compression.buffer_index == m.buffer:
+      d["_compression_metadata"] = compression.unpack()
+
+    entries.append(d)
+
+  return entries
+
+
+def unpack_lut_metadata(lut_tensors):
+  return [{
+      "tensor": t.tensor,
+      "value_buffer": t.valueBuffer,
+      "index_bitwidth": t.indexBitwidth,
+  } for t in sorted(lut_tensors, key=lambda x: x.tensor)]
+
+
+def is_compressed_buffer(buffer_index, unpacked_metadata):
```

Comment:
Method not used.  I believe this was part of the code I added to show compressed bitstrings as lists of indices.

Reply:
Fixed. See 2d56eef.